pkg/pool-manager/pool_test.go

```diff
@@ -706,6 +706,21 @@ var _ = Describe("Pool", func() {
 				Expect(poolManager.macPoolMap).To(HaveLen(1), "macPoolMap should hold the mac address waiting for approval")
 				Expect(poolManager.macPoolMap[allocatedMac]).To(Equal(expectedMacEntry), "macPoolMap's mac's entry should be as expected")
 			})
+			Context("and creating a dry run VM", func() {
+				BeforeEach(func() {
+					newVMDryRun := masqueradeVM.DeepCopy()
+					newVMDryRun.Name = "newVMDryRun"
+
+					By("Create a dry run VM")
+					transactionTimestamp = updateTransactionTimestamp(0)
+					err := poolManager.AllocateVirtualMachineMac(newVM, &transactionTimestamp, false, logger)
+					Expect(err).ToNot(HaveOccurred(), "should successfully create dry run VM")
+				})
+				It("should not storage the dry run mac in the mac pool", func() {
+					Expect(poolManager.macPoolMap).To(HaveLen(1), "macPoolMap should hold only the mac address waiting for approval")
```

Comment:
can we compare the entire `poolManager.macPoolMap` before and after?

Reply:
```
Expect(poolManager.macPoolMap).To(HaveLen(1), "macPoolMap should hold only the mac address waiting for approval")
Expect(poolManager.macPoolMap[allocatedMac]).To(Equal(expectedMacEntry), "macPoolMap's mac's entry should be as expected")
```
We basically comparing the actual versus the expecter as a whole,
since there is 1 entry and we compare the value of that,
same as other tests are doing, and they compare what the real VM which is created did to the pool

i don't think its needed
if you insist let me know please with how you want to do that, to add those two lines also in the beginning of the before each ?